storage/posix/files.go

```diff
@@ -418,6 +429,32 @@ type treeState struct {
 	Root []byte `json:"root"`
 }
 
+// ensureVersion will fail if the compatibility version stored in the state directory
+// is not the expected version. If no file exists, then it is created with the expected version.
+func (s *Storage) ensureVersion(version uint16) error {
+	versionFile := filepath.Join(s.path, stateDir, "version")
+
+	if _, err := os.Stat(versionFile); errors.Is(err, os.ErrNotExist) {
+		klog.V(1).Infof("No version file exists, creating")
+		data := make([]byte, 2)
+		binary.LittleEndian.PutUint16(data, version)
+		if err := createExclusive(versionFile, data); err != nil {
+			return fmt.Errorf("failed to create version file: %v", err)
+		}
```

Comment:
Could pop a `return nil` in here, and then lose the final `else` and out-dent that code as the "happy path"?

Reply:
Done.